goa v2: extend not copying over fields

See public gist:
https://gist.github.com/psschroeter/e6c50bbbb64f9e34c1f4326115ee311f

I'm creating a ResultType SomeItemExt that extends SomeItem then returning that for a method. Generate completes without error but generating the example then building the cmd results in the following error:
gen/http/dummy/server/encode_decode.go:25:9: res.IsNotModified undefined (type *dummysvc.MyorgSomeItem has no field or method IsNotModified)

Looking at openapi.json, it's missing the result types from there.
For the show dummy method, responses is:
```
        "responses": {
          "200": {
            "description": "OK response."
          },
          "304": {
            "description": "Not Modified response."
          }
        },
```

But should a reference to the result type like in there somehwere..
```
            "schema": {
              "$ref": "#/definitions/MyorgSomeItem..."
            }
```

--- a/design/result_type.go
+++ b/design/result_type.go
@@ -179,7 +179,8 @@ func (m *ResultTypeExpr) ComputeViews() []*ViewExpr {
 	return nil
 }
 
-// Finalize builds the default view if not explicitly defined.
+// Finalize builds the default view if not explicitly defined and finalizes
+// the underlying UserTypeExpr.
 func (m *ResultTypeExpr) Finalize() {
 	if m.View("default") == nil {
 		v := &ViewExpr{
@@ -189,6 +190,7 @@ func (m *ResultTypeExpr) Finalize() {
 		}
 		m.Views = append(m.Views, v)
 	}
+	m.UserTypeExpr.Finalize()
 }
 
 // Project creates a ResultTypeExpr containing the fields defined in the view

--- a/dsl/result_type.go
+++ b/dsl/result_type.go
@@ -78,11 +78,13 @@ func ResultType(identifier string, fn func()) *design.ResultTypeExpr {
 	}
 	canonicalID := design.CanonicalIdentifier(identifier)
 	// Validate that result type identifier doesn't clash
-	if m := design.Root.UserType(canonicalID); m != nil {
-		eval.ReportError(
-			"result type %#v with canonical identifier %#v is defined twice",
-			identifier, canonicalID)
-		return nil
+	for _, rt := range design.Root.ResultTypes {
+		if re := rt.(*design.ResultTypeExpr); re.Identifier == canonicalID {
+			eval.ReportError(
+				"result type %#v with canonical identifier %#v is defined twice",
+				identifier, canonicalID)
+			return nil
+		}
 	}
 	identifier = mime.FormatMediaType(identifier, params)
 	lastPart := identifier

